fedn/cli/run_cmd.py

```diff
@@ -163,11 +167,11 @@ def build_cmd(ctx, path):
     dispatcher = Dispatcher(config, path)
     _ = dispatcher._get_or_create_python_env()
     dispatcher.run_cmd("build")
-
-    # delete the virtualenv
-    if dispatcher.python_env_path:
-        logger.info(f"Removing virtualenv {dispatcher.python_env_path}")
-        shutil.rmtree(dispatcher.python_env_path)
+    if venv:
+        # delete the virtualenv
+        if dispatcher.python_env_path:
```

Comment:
what happens if the dispatcher lack a python_env_path? Perhpas capture that behavior with logging.